client-runtime/src/main/java/com/microsoft/rest/RestProxy.java

```diff
@@ -136,25 +136,38 @@ private HttpRequest createHttpRequest(SwaggerMethodParser methodParser, Object[]
         final Object bodyContentObject = methodParser.body(args);
         if (bodyContentObject != null) {
             String contentType = methodParser.bodyContentType();
+            if (contentType == null || contentType.isEmpty()) {
+                contentType = request.headers().value("Content-Type");
+            }
+            if (contentType == null || contentType.isEmpty()) {
+                if (bodyContentObject instanceof byte[] || bodyContentObject instanceof String) {
+                    contentType = ContentType.APPLICATION_OCTET_STREAM;
+                }
+                else {
+                    contentType = ContentType.APPLICATION_JSON;
+                }
+            }
+
+            request.headers().set("Content-Type", contentType);
+
             if (ContentType.APPLICATION_JSON.equalsIgnoreCase(contentType)) {
                 final String bodyContentString = serializer.serialize(bodyContentObject);
                 request.withBody(bodyContentString, contentType);
             }
             else if (bodyContentObject instanceof byte[]) {
-                if (contentType == null) {
-                    contentType = ContentType.APPLICATION_OCTET_STREAM;
-                }
                 request.withBody((byte[]) bodyContentObject, contentType);
             }
             else if (bodyContentObject instanceof String) {
-                if (contentType == null) {
-                    contentType = ContentType.APPLICATION_OCTET_STREAM;
+                final String bodyContentString = (String)bodyContentObject;
+                if (!bodyContentString.isEmpty()) {
+                    request.withBody((String) bodyContentObject, contentType);
                 }
-                request.withBody((String) bodyContentObject, contentType);
             }
             else {
                 final String bodyContentString = serializer.serialize(bodyContentObject);
-                request.withBody(bodyContentString, contentType);
+                if (!bodyContentString.isEmpty()) {
```

Comment:
If you are able to get the serializer to produce an empty string, let me know, because it can induce some shaky server-side behaviors when an empty/null body is interpreted as JSON. Even a null `bodyContentObject` should be turned into the string `null` by the serializer.

Taking the browser's JSON serializer as a reference implementation:
```javascript
> JSON.stringify('')
< "\"\""
> JSON.stringify(null)
< "null"
```

Reply:
Also:
```javascript
> JSON.parse('')
< SyntaxError: JSON Parse error: Unexpected EOF
```
Some server-side body parsers will interpret it as an empty object, but they reasonably have the right to get upset about being asked to parse an empty body into JSON. The language of "parse an empty string" versus "parse a string containing a pair of quotes" is hard to keep straight in this discussion..